Dockerfile

```diff
@@ -0,0 +1,45 @@
+####################################################
+# Use python 2.x and the latest version of alpine  #
+####################################################
+FROM python:2-alpine
+
+LABEL maintainer=peter@grainger.xyz
+
+###################################################
+# Add a compiler                                  #
+###################################################
+RUN apk update && \
+    apk add --no-cache gcc \
+                       libffi-dev \
+                       openssl-dev \
+                       libxslt-dev \
+                       libxml2-dev \
+                       musl-dev \
+                       linux-headers
+
+###################################################
+# Create somewhere to put the files               #
+###################################################
+RUN mkdir -p /opt/jarvis
+WORKDIR /opt/jarvis
+
+###################################################
+# Requirements file first to help cache           #
+###################################################
+COPY requirements.txt .
```

Comment:
You should ideally always place your requirements before the code base. This will prevent re-installing the requirements if a small part of the source code changes. Doesn't make a lot of difference but saves plenty of build time.

Reply:
Sorry I didnt pay attention to the `COPY ./ ./`, my bad 😞